injected/src/features/message-bridge/create-page-world-bridge.js

```diff
@@ -0,0 +1,134 @@
+import {
+    InstallProxy,
+    ProxyNotification,
+    ProxyRequest,
+    ProxyResponse,
+    SubscriptionRequest,
+    SubscriptionResponse,
+    SubscriptionUnsubscribe,
+} from './schema.js';
+
+/**
+ * todo: use the already-captured ones
+ */
+const captured = {
+    dispatchEvent: globalThis.dispatchEvent.bind(globalThis),
+    addEventListener: globalThis.addEventListener.bind(globalThis),
+    removeEventListener: globalThis.removeEventListener.bind(globalThis),
+    CustomEvent: globalThis.CustomEvent,
+    Promise: globalThis.Promise,
+    String: globalThis.String,
+    mathRandom: globalThis.Math.random.bind(globalThis.Math),
+};
+
+/**
+ * This part can be called from any script.
+ * @param {string} featureName
+ * @param {string} token
+ * @return {Pick<import("@duckduckgo/messaging").Messaging, 'notify' | 'subscribe' | 'request'>}
+ */
+export function createPageWorldBridge(featureName, token) {
+    console.log('will createPageWorldBridge for', featureName, 'with token', token);
+
+    /**
+     * TODO: what is the value of this token? - since 'createPageWorldBridge' is available to all page scripts, probably none
+     * @param {string} eventName
+     * @return {`${string}-${string}`}
+     */
+    function appendToken(eventName) {
+        return `${eventName}-${token}`;
+    }
```

Comment:
i'm not tool convinced there's any value in using a token across both ends of the proxy - even one that's safe. Because in the end we're still exposing the API to the page world - meaning scripts call `navigator.duckduckgo.createMessagingBridge()`. The main difference this PR introduces, is that it allows messaging without **directly** exposing the webkit handlers to the page - it's not here to be 100% secure, unfortunately

Reply:
I don't see much harm in keeping the method to wrap it with a string that reduces the chance of colliding with pages.

I'd rather we not use the session token as it might leak.

We could use a derived key from it but as you mention the security doesn't matter that much (we need to gate the client side)

Reply:
aka a prefix of `ddg-something-message-something-` is probably more useful than a unique identifier.

The unique token would reduce the pages ability to snoop on the message contents as well.